### ISSUE TITLE
Implementation Issue

### ISSUE THREAD

**Issue by Saumya-Mishra9129** [sugarlabs/showntell-activity#17]
> gi.repository.GLib.Error: g-invoke-error-quark: Class slideviewer+ThumbViewer doesn't implement draw 

Tested with master branch.

**Comment by quozl** [comment]
Indeed.  You can see why, I hope;

* https://github.com/sugarlabs/showntell-activity/blob/master/slideviewer.py#L281 does declare a signal may be emitted by the class,
* there is no call to emit that signal,
* therefore the signal need not be declared.

**Comment by Saumya-Mishra9129** [comment]
Yeah got it. I have fixed it in dae1f6c in #16

**Comment by quozl** [comment]
Where is configure_event emitted?

**Comment by Saumya-Mishra9129** [comment]
> Where is configure_event emitted?

Here-https://github.com/sugarlabs/showntell-activity/blob/ba1ddd3c3cf0d3cc1cd39a33bd13d10f0be012a8/slideviewer.py#L239
Thanks for pointing this out, I guess  dae1f6c is not a correct way to solve this. We are also using here a method "_do_draw_cb" , the issue is related to that I guess. I have to test once by changing method name.

**Comment by quozl** [comment]
No, that's not a signal emission.  A signal is emitted like this;

```
    self.emit('signal-name')
```

A GObject derived class has to define the signal, and emit it in some way at an appropriate time.  If the calling program doesn't connect to the signal by name, then the emit has no effect.

Part of https://github.com/sugarlabs/showntell-activity/commit/dae1f6cdeebaa94e3037f1ee68819743ce312e94 seems correct; the removal of the unimplemented signal in the ThumbViewer class, as that is mentioned in the error message.  But the change to SlideViewerCanvas is unexpected, and let me see that this class too doesn't emit a signal.  It looks like the author was misled about signals, and that the same mistake is repeated.  It would be best to survey all use of signals in this activity to pre-empt the next error report.

**Comment by Saumya-Mishra9129** [comment]
> It looks like the author was misled about signals, and that the same mistake is repeated. It would be best to survey all use of signals in this activity to pre-empt the next error report.

Thanks, Sure

**Comment by Saumya-Mishra9129** [comment]
@quozl Just out of curiousity what does this override means here
https://github.com/sugarlabs/showntell-activity/blob/ba1ddd3c3cf0d3cc1cd39a33bd13d10f0be012a8/slideviewer.py#L34
Does it mean that we are overriding the given signal.?

**Comment by quozl** [comment]
Don't know, sorry.  I suggest you search in the source code, https://github.com/GNOME/pygobject or https://gitlab.gnome.org/GNOME/pygobject.

(GSoC mentor comment; you can certainly spend paid student time researching; I'd budget anything between five and twelve hours to resolve this question.  You may need to search, review source code, learn minimal C, and test.)
